Various v6 breaking changes (#741)
mongodb/docs-node: source/whats-new.txt

     import these methods from the driver instead of from
     ``mongodb-client-encryption``.

   - Removed support for strings with a length of 12 from the ``ObjectId``
     constructor.

>    - The ``abortTransaction`` and ``commitTransaction`` methods return

Reviewer: Suggestion:
Some entries use the "()" suffix to identify methods names. I think it helps make certain less ambiguous and would make sense to make these consistent.

     ``null`` instead of the raw command results.

>    - Removed connection option helpers that recognized values other than ``true``

Reviewer: Suggestion:

While this style guide guidance is sometimes borderline, it mentions [don't humanize inanimate objects](https://www.mongodb.com/docs/meta/style-guide/writing/use-effective-verbs/#don-t-humanize-inanimate-objects). I think it is not common to describe methods as performing the "recognize" action:

```suggestion
   - Removed connection option helpers that accepted values other than ``true``
```

Other alternatives could be "parsed" or "casted"

>      or  ``false`` as booleans. You must provide either ``true`` or ``false`` values for
>      options that expect boolean values in the connection string or MongoClient
>      constructor.

Reviewer: Suggestion:

I think this might be a more direct way to explain the new requirement:

```suggestion
     or  ``false`` as booleans. You must provide either ``true`` or ``false`` values in
     the connection string or to the MongoClient constructor.
```

   - Removed the Binary BSON type constructor that accepted a string.

   - The ``Binary.write`` method no longer accepts a string to write

>      to the binary BSON object.

Author: I'm not sure if the suggested change is correct since they didn't remove the method that accepted the string. They just removed string as one of the accepted parameters to that method

>    - ``ClientEncryption`` APIs return promises instead of callbacks.

Reviewer: Suggestion:

I think this should be changed to be consistent: https://github.com/mongodb/docs-node/pull/741#discussion_r1296520856

Author: Oops, looks like I pushed without saving that change. Fixed
